src/store/traction/ont/runs/mutations.js

```diff
@@ -0,0 +1,46 @@
+import Vue from 'vue'
+
+const mutate = (key) => (state, val) => {
+  state[key] = val
+}
+
+const mutateRun = (key) => (state, val) => {
+  state.currentRun[key] = val
+}
+
+const setFlowcellId = () => (state, obj) => {
+  const exists = state.currentRun.flowcell_attributes.find(
+    (flowcell) => flowcell.position == obj.position,
+  )
+
+  if (exists) {
+    Vue.set(exists, 'flowcell_id', obj.$event)
+  } else {
+    const flowcell = { ...{ flowcell_id: obj.$event }, ...{ position: obj.position } }
```

Comment:
Not sure what the destructuring does here

Reply:
Hm do you mean the `...`? This is creating a object using the spread operator (`...`) to create a new object from other objects

Reply:
Oh I'm with you. It could be something simply like `{ flowcell_id: obj.$event, position: obj.position }` ?

Reply:
Yes done👍 cheers